components/infobox/wikis/starcraft/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,510 @@
+	player.calculateEarnings = CustomPlayer.calculateEarnings
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+	player.getWikiCategories = CustomPlayer.getWikiCategories
+	player.getPersonType = CustomPlayer.getPersonType
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'status' then
+		return {
+			Cell{
+				name = 'Race',
+				content = {CustomPlayer._getRaceData(_args.race or 'unknown', _RACE_FIELD_AS_CATEGORY_LINK)}
+			}
+		}
+	elseif id == 'role' then return {}
+	elseif id == 'region' then return {}
+	elseif
+		id == 'history' and
+		string.match(_args.retired or '', '%d%d%d%d')
+	then
+		table.insert(widgets, Cell{
+				name = 'Retired',
+				content = {_args.retired}
+			})
+	end
+	return widgets
+end
+
+function CustomInjector:addCustomCells(widgets)
+	--enable this AFTER the match to match2 conversion of 1v1 matchlists and brackets
+	--local yearsActive = Namespace.isMain() and CustomPlayer._getMatchupData() or nil
+
+	local currentYearEarnings = _earningsGlobal[tostring(_CURRENT_YEAR)]
+	if currentYearEarnings then
+		currentYearEarnings = Math.round{currentYearEarnings}
+		currentYearEarnings = '$' .. mw.language.new('en'):formatNum(currentYearEarnings)
+	end
+
+	return {
+		Cell{
+			name = 'Approx. Winnings ' .. _CURRENT_YEAR,
+			content = {currentYearEarnings}
+		},
+		--Cell{name = 'Years active', content = {yearsActive}}
+	}
+end
+
+function CustomPlayer.nameDisplay()
+	CustomPlayer._getRaceData(_args.race or 'unknown')
+	local raceIcon = RaceIcon.getNormalIcon{_raceData.race}
+	local name = _args.id or _player.pagename
+
+	return raceIcon .. '&nbsp;' .. name
+end
+
+function CustomPlayer._getRaceData(race, asCategory)
+	race = string.lower(race)
+	race = CleanRace[race] or race
+	local raceTable = _RACE_DATA[race]
+
+	local faction, faction2
+	if race == _RACE_ALL_SHORT then
+		faction = _RACE_ALL
+	else
+		faction = (raceTable or {})[1]
+		faction2 = (raceTable or {})[2]
+	end
+
+	local display
+	if not raceTable and race ~= 'unknown' then
+		display = '[[Category:InfoboxRaceError]]<strong class="error">' ..
+			mw.text.nowiki('Error: Invalid Race') .. '</strong>'
+	else
+		if asCategory then
+			for raceIndex, raceValue in ipairs(raceTable or {}) do
+				raceTable[raceIndex] = ':Category:' .. raceValue .. ' Players|' .. raceValue .. ']]'
+					.. '[[Category:' .. raceValue .. ' Players'
+			end
+		end
+		if raceTable then
+			display = '[[' .. table.concat(raceTable, ']],&nbsp;[[') .. ']]'
+		end
+	end
+
+	_raceData = {
+		race = race,
+		faction = faction or '',
+		faction2 = faction2 or '',
+		display = display,
+	}
+
+	return display
+end
+
+function CustomPlayer.adjustLPDB(_, lpdbData)
+	_lpdbData = lpdbData
+	local extradata = lpdbData.extradata or {}
+	extradata.race = _raceData.race
+	extradata.faction = _raceData.faction
+	extradata.faction2 = _raceData.faction2
+	extradata.lc_id = string.lower(_player.pagename)
```

Comment:
This is lowerCase pageName, not lowerCase id?

Reply:
forgot to kick it altogether